### PR TITLE
SONARJAVA-1016 RSPEC-864 Update implementation to remove issues

### PR DESCRIPTION
**Pull request by merciesa**
- Remove requirements between arithmetic operators
- Remove requirement of parenthesis on ternary operands

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/OperatorPrecedenceCheck.java [hunk 1]
```diff
@@ -119,20 +121,10 @@ public void visitArrayAccessExpression(ArrayAccessExpressionTree tree) {
...
-    stack.pop();
-  }
-
-  @Override
   public void visitBinaryExpression(BinaryExpressionTree tree) {
-    Tree.Kind kind = getKind(tree);
     Tree.Kind peek = stack.peek();
-    if (peek != null && peek != kind && !isException(peek, kind)) {
+    Tree.Kind kind = getKind(tree);
+    if (TABLE.get(peek, kind) == Boolean.TRUE) {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/OperatorPrecedenceCheck.java]: :new: MAJORChange this comparison to use the equals method.[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1698)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 new issue:
* 1 major

Details:
* [Change this comparison to use the equals method.](https://github.com/SonarSource/blob/98daacdba37a273149443f3eb27dec61df99c838/java-checks/src/main/java/org/sonar/java/checks/OperatorPrecedenceCheck.java#L127)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1698)

File: java-checks/src/main/java/org/sonar/java/checks/OperatorPrecedenceCheck.java (hunk 1 above)
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/OperatorPrecedenceCheck.java]: This should use BooleanUtils.isTrue.

File: java-checks/src/main/java/org/sonar/java/checks/OperatorPrecedenceCheck.java [hunk 2]
```diff
@@ -56,37 +56,39 @@
...
-    Tree.Kind.MINUS_ASSIGNMENT,
-    Tree.Kind.MULTIPLY_ASSIGNMENT,
-    Tree.Kind.OR_ASSIGNMENT,
-    Tree.Kind.PLUS_ASSIGNMENT,
-    Tree.Kind.REMAINDER_ASSIGNMENT,
-    Tree.Kind.RIGHT_SHIFT_ASSIGNMENT,
-    Tree.Kind.UNSIGNED_RIGHT_SHIFT_ASSIGNMENT,
-    Tree.Kind.XOR_ASSIGNMENT
-    );
+  private static final Table<Tree.Kind, Tree.Kind, Boolean> TABLE;
```
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/OperatorPrecedenceCheck.java]: Name is way too generic.

**Comment by SonarTech** [comment]
SonarQube analysis reported no new issues.

**Comment by benzonico** [comment]
LGTM

**Pull request closed without merging**